wallet-new/src/Cardano/Wallet/Kernel.hs

```diff
@@ -151,10 +151,9 @@ initPassiveWallet logMessage keystore handles node = do
                 , _walletMeta            = hMeta handles
                 , _walletNode            = node
                 , _walletSubmission      = submission
+                , _walletSubmissionLock  = submissionLock
```

Comment:
There is no need for an additional lock here, is there? The submission state _itself_ is already an `MVar`, and juggling multiple locks is a dangerous game (take them in the wrong order and end up with deadlock). I'd prefer we just stick with a single lock, unless there is a good reason not to.

Reply:
@edsko that was my first approach but it`s not exactly the same 
```
        tickFunction :: MVar () -> MVar WalletSubmission -> IO ()
        tickFunction lock submissionLayer = do
            modifyMVar_ lock $ \_ -> do
                (cancelled, toSend) <-
                    modifyMVar submissionLayer $ \layer -> do
                        let (e, s, state') = tick layer
                        return (state', (e,s))
                unless (Map.null cancelled) $
                    cancelPending walletPassive cancelled
                sendTransactions toSend
                return ()
``` 

`cancellPending` and `sendTransaction` is called withut the existing MVar. If we want to use the existing MVar as lock, we have to move them inside the modifyMVar IO action. I guess that`s fine to do? I don`t think it will change much.

Reply:
I think we can even just leave this as is. If we take the MVar elsewehere, this function won't be able to get `cancelled` and `toSend`. If it just so happens that it just computed those at the point where we call reset, the worst that can happen is that it will send out some transactions (which is fine), and remove some transactions from the DB (but they won't be there anymore anyway).

Reply:
ok, having a pending cancelPending action may be safe on an empty db, but what if after the reset, some new txs are inserted? We have no guarantee that the cancelPending action will happen before the insertion. Am I getting paranoid?

Reply:
Hmm, fair point. Ok, let's move the `cancelPending` inside the `modifyMVar` maybe? And document the reason for this.